docs/vgclog.md

```diff
@@ -0,0 +1,423 @@
+
+**Note:** You can run one or more GC traces by using the (['-XTgc' option](xtgc.md)) to get more detailed information to help diagnose GC problems or perform finer tuning.
+
+## How to generate a verbose garbage collection log 
+
+You can enable verbose GC logs by specifying the `-verbose:gc` standard option when you start your application. For more information, see [standard command-line options](cmdline_general.md).
+
+The output of `-verbose:gc` is printed to STDERR by default. To print the log output to a file, append the `-Xverbosegclog` option. You can print the verbose GC log file to a succession of files, where each file logs a specified number of GC cycles, by using additional parameters of `-Xverbosegclog` option. For more information, see the command-line option [-Xverbosegclog](xverbosegclog.md).
+
+## Verbose GC log contents 
+
+The Verbose garbage collection logs are printed in XML format and consist of the following sections:  
+
+- a summary of your GC configuration, which is captured in the stanza beginning with the [`<INITIATIZED>` tag](vgclog.md#xml-tags-an-attributes) 
+
+- stanzas that contain information about the different garbage collection cycles run, including the GC increments and GC operations that made up the GC cycle. 
+
+The verbose GC logs are event-based, recording each *stop-the-world* event as it happens. While *stop-the-world* cycles are fully logged, some aspects of concurrent cycles are also logged, even though operations that are running concurrently with application threads are not logged. Concurrent operations run as part of a concurrent cycle, and these concurrent cycles begin and finish with *stop-the-world* events - events that are logged. (For some garbage collectors, the concurrent cycle also includes intermediate *stop-the-world* events). In addition, for some types of concurrent collections, verbose GC logs also include information about when certain targets are reached. 
+
+The highest level stanzas in the verbose GC log represent GC cycles and begin with xml tags that define the kickoff event and reason. Except for some of the more complex GC algorithms (such as the global marking phase GC of the balanced policy), a GC cycle, when complete, generally results in reclaimed memory for reuse. Each cycle consists of 1 or more GC increments and GC operations.  
+
+Events that make up a garbage collector cycle are listed in the stanza in a linear fashion, but are ‘nested’ by a start and end tag that identify the start and end of an event. Details of the event are recorded in the XML attributes associated with the tag and also within tags that are nested within the GC event's start and end tags.
+
+The following table shows the tagging that is used for the nested GC processes: 
```

Comment:
I'm wondering if it also makes sense to lay out the tagging model, devoid of information, so that users can understand what gets nested where. This looks to be something like:

```
<exclusive-start>
    <af-start>
        <cycle-start>
            <gc-start> </gc-start>
                 <gc-op> </gc-op>
                 <gc-op> </gc-op>
            <gc-end> </gc-end>
            <gc-start> </gc-start>
                 <gc-op> </gc-op>
                 <gc-op> </gc-op>
            <gc-end> </gc-end>
        <cycle-end>
    <af-end>
<exclusive-end>
```

The table is still good information, but this might also help to frame the layout in people's minds. I was initially thrown by references to start and end tags ..... E.g. `<gc-op>` does have an end `</gc-op>`, even though there is no `<gc-op-start>` etc tag.

Reply:
Really like this Sue, I'll work on getting that added.

Reply:
With that high level view, you can try to present a STW GC increment. exclusive-start/end and gc-start/end are typically part of a such increment. 

Another part of increment is the event that triggered: af-start in this case . Some other triggers are possible such as: concurrent-global-final or allocation-taxation (used in Balanced). 

cycle-start and cycle-end are optional part of increment. They say it's the first and final GC increment of a cycle respectively or the only increment, if both of them are reported.

Reply:
Part of the new draft.